comp/logs/agent/config/config.go

```diff
@@ -172,6 +172,36 @@ func buildTCPEndpoints(coreConfig pkgconfigmodel.Reader, logsConfig *LogsConfigK
 	}
 
 	additionals := loadTCPAdditionalEndpoints(main, logsConfig)
+
+	// Add in the MRF endpoint if MRF is enabled.
+	if coreConfig.GetBool("multi_region_failover.enabled") {
+		mrfURL, err := pkgconfigutils.GetMRFLogsEndpoint(coreConfig, tcpEndpointPrefix)
+		if err != nil {
+			return nil, fmt.Errorf("cannot construct MRF endpoint: %s", err)
+		}
+
+		mrfHost, mrfPort, mrfUseSSL, err := parseAddressWithScheme(mrfURL, defaultNoSSL, parseAddressAsHost)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse %s: %v", mrfURL, err)
+		}
+
+		e := NewEndpoint(coreConfig.GetString("multi_region_failover.api_key"), "multi_region_failover.api_key", mrfHost, mrfPort, mrfUseSSL)
```

Comment:
Does it make sense ot use `newTCPEndpoint` here? 
If so I think you can slim down the code below applying all the overrides.

Reply:
Ah yes, that would make sense.

Reply:
After looking at it, I'm actually not sure it would make that much sense. The `newTCPEndpoint()` function seems to take a `logsConfigKeys` object as it's object, however the MRF endpoints don't use the `logs_config` preface so I don't think they wouldn't be available in the this object. The additional TCP endpoints don't use this function either: https://github.com/DataDog/datadog-agent/blob/main/comp/logs/agent/config/endpoints.go#L149C18-L149C19

Reply:
FWIW `logsConfigKeys` can be configured wiht a prefix, so it's not just for `logs_config` configs. 
But copying it from `main` like you already are is fine. Was just trying to think of ways to reuse code.